telethon/events/__init__.py

```diff
@@ -229,13 +231,15 @@ def build(self, update):
             return
 
         # Short-circuit if we let pass all events
-        if all(x is None for x in (self.incoming, self.outgoing, self.chats)):
+        if all(x is None for x in (self.incoming, self.outgoing, self.chats, self.pattern)):
             return event
 
         if self.incoming and event.message.out:
             return
         if self.outgoing and not event.message.out:
             return
+        if self.regex and not re.match(self.regex, event.raw_text):
```

Comment:
`self.regex` instead `self.pattern`. Are you sure it works?

Reply:
Also please don't use `event.raw_text`, access `event.message.message` instead (I'd rather avoid accessing properties when building the event).

Reply:
sorry, I created it in my test version and forgot to rename everything. this should be self.pattern as well